server/routes/activities/create-an-activity/handlers/removeEndDate.test.ts

```diff
@@ -0,0 +1,163 @@
+import { Request, Response } from 'express'
+import { plainToInstance } from 'class-transformer'
+import { validate } from 'class-validator'
+import { when } from 'jest-when'
+import { associateErrorsWithProperty } from '../../../../utils/utils'
+import RemoveEndDateRoutes, { RemoveEndDateOptions } from './removeEndDate'
+import ActivitiesService from '../../../../services/activitiesService'
+import atLeast from '../../../../../jest.setup'
+import activity from '../../../../services/fixtures/activity_1.json'
+import { Activity } from '../../../../@types/activitiesAPI/types'
+
+jest.mock('../../../../services/activitiesService')
+
+const activitiesService = new ActivitiesService(null) as jest.Mocked<ActivitiesService>
+
+describe('Route Handlers - Create an activity schedule - Remove end date', () => {
+  const handler = new RemoveEndDateRoutes(activitiesService)
+  let req: Request
+  let res: Response
+
+  beforeEach(() => {
+    res = {
+      locals: {
+        user: {
+          username: 'joebloggs',
+        },
+      },
+      render: jest.fn(),
+      redirectOrReturn: jest.fn(),
+      redirectWithSuccess: jest.fn(),
+    } as unknown as Response
+
+    req = {
+      session: {
+        createJourney: {},
+      },
+    } as unknown as Request
+  })
+
+  describe('GET', () => {
+    it('should render the expected view', async () => {
+      await handler.GET(req, res)
+      expect(res.render).toHaveBeenCalledWith('pages/activities/create-an-activity/remove-end-date')
+    })
+  })
+
+  describe('POST', () => {
+    it('edit - should save selected option in session and redirect to end date edit page', async () => {
+      req.body = {
+        removeEndDate: 'change',
+      }
+
+      req.session.createJourney = {
+        activityId: 1,
+        name: 'Maths level 1',
+      }
+
+      req.params = {
+        mode: 'edit',
+      }
+
+      await handler.POST(req, res)
+
+      expect(req.session.createJourney.removeEndDate).toEqual('change')
```

Comment:
Might want to check `session.createJourney.endDate` has not been altered

Reply:
added check